valhalla/mjolnir/traffic_updater.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <array>
+#include <cstdint>
+#include <future>
+#include <optional>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include <boost/property_tree/ptree.hpp>
+
+#include "baldr/graphid.h"
+#include "baldr/predictedspeeds.h"
+#include "filesystem.h"
+
+namespace valhalla {
+namespace mjolnir {
+
+// Struct to hold stats information during each threads work
+struct TrafficStats {
+  uint32_t constrained_count = 0;
+  uint32_t free_flow_count = 0;
+  uint32_t compressed_count = 0;
+  uint32_t updated_count = 0;
+  uint32_t dup_count = 0;
+
+  // Accumulate counts from all threads
+  void operator()(const TrafficStats& other);
+};
+
+struct TrafficSpeeds {
+  uint8_t constrained_flow_speed = 0;
+  uint8_t free_flow_speed = 0;
+  std::optional<std::array<int16_t, valhalla::baldr::kCoefficientCount>> coefficients;
+};
+
+// Functions declarations
+std::unordered_map<uint32_t, TrafficSpeeds>
+ParseTrafficFile(const std::vector<std::string>& filenames, TrafficStats& stat);
+
+void UpdateTile(const std::string& tile_dir,
+                const valhalla::baldr::GraphId& tile_id,
+                const std::unordered_map<uint32_t, TrafficSpeeds>& speeds,
+                TrafficStats& stat);
+
+void UpdateTiles(
+    const std::string& tile_dir,
+    std::vector<std::pair<valhalla::baldr::GraphId, std::vector<std::string>>>::const_iterator
+        tile_start,
+    std::vector<std::pair<valhalla::baldr::GraphId, std::vector<std::string>>>::const_iterator
+        tile_end,
+    std::promise<TrafficStats>& result);
+
+std::vector<std::pair<valhalla::baldr::GraphId, std::vector<std::string>>>
+PrepareTrafficTiles(const filesystem::path& traffic_tile_dir);
```

Comment:
for all of the header level functions if they are in public headers we like to put java doc so that at a glance people can understand what the method is doing and what he arguments mean etc. can you add that to all of these. also if any of them could be moved to just the implementation (ie not exposed in the public library api) that would be best. the less stuff we have in headers the less of a headache the library is to maintain and to work with for people who use it that way